.github/scripts/transform_dbt_test_results.py

```diff
@@ -0,0 +1,1082 @@
+    """Load a list of TestCategory objects from a cache located at
+    `file_path`."""
+    with open(file_path) as cache_file:
+        test_category_dicts = json.load(cache_file, use_decimal=True)
+    return [
+        TestCategory.from_dict(category_dict)
+        for category_dict in test_category_dicts
+    ]
+
+
+def save_test_categories_to_file(
+    test_categories: typing.List[TestCategory], file_path: str
+) -> None:
+    """Save a list of TestCategory objects to a cache located at
+    `file_path`."""
+    test_category_dicts = [
+        test_category.to_dict() for test_category in test_categories
+    ]
+    os.makedirs(TEST_CACHE_DIR, exist_ok=True)
+    with open(file_path, "w") as cache_file:
+        json.dump(test_category_dicts, cache_file, use_decimal=True)
+
+
+def get_test_categories_from_athena(
+    run_results_filepath: str, manifest_filepath: str
+) -> typing.List[TestCategory]:
+    """Load a list of TestCategory objects by querying Athena for
+    test results generated from a `dbt test --store-failures` call."""
+    with open(run_results_filepath) as run_results_fobj:
+        run_results = json.load(run_results_fobj)
+
+    with open(manifest_filepath) as manifest_fobj:
+        manifest = json.load(manifest_fobj)
+
+    test_categories = get_test_categories(run_results, manifest)
+    if not test_categories:
+        raise ValueError(f"{run_results_filepath} contains no test results")
+
+    return test_categories
+
+
+def get_test_categories(
+    run_results: typing.Dict, manifest: typing.Dict
+) -> typing.List[TestCategory]:
+    """Given two artifacts from a `dbt test --store-failures` call (a
+    run_results.json file dict and a manifest.json file dict), generates a list
+    of TestCategory objects storing the results of the tests."""
+    conn = pyathena.connect(
+        s3_staging_dir=AWS_ATHENA_S3_STAGING_DIR,
+        region_name="us-east-1",
+        cursor_class=pyathena.cursor.DictCursor,
+    )
+    cursor = conn.cursor()
+
+    tests_by_category: typing.Dict[str, TestCategory] = {}
+
+    for run_result in run_results["results"]:
+        unique_id = run_result["unique_id"]
+        node = manifest["nodes"].get(unique_id)
+        if node is None:
+            raise ValueError(f"Missing dbt manifest node with id {unique_id}")
+
+        test_name = node["name"]
+        status = run_result["status"]
+        execution_time = run_result["execution_time"]
+
+        meta = node.get("meta", {})
+        category = get_category_from_node(node)
+        tablename = get_tablename_from_node(node)
+        test_description = meta.get("description")
+
+        # Basic attrs for the test result that apply whether or not the test
+        # failed
+        base_result_kwargs = {
+            "name": test_name,
+            "table_name": tablename,
+            "description": test_description,
+            "elapsed_time": execution_time,
+        }
+
+        if status == Status.PASS.value:
```

Comment:
The previous version of this function, [`get_test_results_by_category`](https://github.com/ccao-data/data-architecture/blob/66c85bfa973d163bd29c4e41f08ecf58eafff08d/.github/scripts/format_dbt_test_failures.py#L402), only supported tests with status `fail`. With the addition of this block, we now support passing tests as well.

Reply:
Same question as above: what happens for `WARNING` tests?

Reply:
Answered here: https://github.com/ccao-data/data-architecture/pull/370#discussion_r1554070863